### ISSUE TITLE
Clarification about the splits used in the training time 

### ISSUE THREAD

**Issue by danyaljj** [google-research/text-to-text-transfer-transformer#102]
Hi there, 

Wanted get some clarification about what splits are used during fine-tuning. 
In particular, when calling the following command: 

```python 
    print(" >>>>>>> fine-tuning  . . . ")
    model.finetune(
        mixture_or_task_name="my_questions",
        pretrained_model_dir=PRETRAINED_DIR,
        finetune_steps=FINETUNE_STEPS
    )
```
and given the following task registry definition: 
```python 
t5.data.TaskRegistry.add(
    "my_task",
    # Supply a function which returns a tf.data.Dataset.
    dataset_fn=dataset_fn,
    splits=["split1", "split2", "split3"],
    # Supply a function which preprocesses text from the tf.data.Dataset.
    text_preprocessor=[dataset_preprocessor],
    # Use the same vocabulary that we used for pre-training.
    sentencepiece_model_path=t5.data.DEFAULT_SPM_PATH,
    # Lowercase targets before computing metrics.
    postprocess_fn=t5.data.postprocessors.lower_text,
    # We'll use accuracy as our evaluation metric.
    metric_fns=[t5.evaluation.metrics.accuracy],
    # Not required, but helps for mixing and auto-caching.
    num_input_examples=num_examples
)
```
does the code use *all* the splits specified in `splits` to fine-tune (regardless of what the names are)? 

Complete code here: https://gist.github.com/danyaljj/288b58427c50f9ef844a767276d430f7

**Comment by adarob** [comment]
Thanks for the question as we may need to document this better. Only
'train' is used for training and 'validation' is used for evaluation. This
can be changed in the gin config if necessary however.

**Comment by adarob** [comment]
Leaving this open so we can add optional kwargs to MtfModel

**Comment by danyaljj** [comment]
Thanks for the clarification (and the efforts to improve the code/documentation)! 

Just a thought about the thread, it's odd that for evaluation you have the option to specify a `split`: 
```
    model.eval(
        mixture_or_task_name="my_questions",
        checkpoint_steps="all",
        split="split3"
    )
```
, while I didn't find such an option for `finetune`.

**Comment by craffel** [comment]
> Just a thought about the thread, it's odd that for evaluation you have the option to specify a split

This is because you might want to eval on different splits - validation or test, for example.

> while I didn't find such an option for finetune.

This is because we assume you always want to train and finetune on the "train" split.

**Comment by craffel** [comment]
This should have been auto-closed by #119 .

**Comment by adarob** [comment]
#119 isn't closed yet!
